packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js

```diff
@@ -0,0 +1,582 @@
+    // Dry matter content is currently 100 - moisture content
+    table.decimal('moisture_content_percent', 36, 12);
+    table.check(
+      `(COALESCE(${elements.join(
+        ', ',
+      )}) IS NULL AND elemental_unit IS NULL) OR (COALESCE(${elements.join(
+        ', ',
+      )}) IS NOT NULL AND elemental_unit IS NOT NULL)`,
+      [],
+      'elemental_unit_check',
+    );
+    table.check(
+      `elemental_unit != 'percent' OR (elemental_unit = 'percent' AND (${elements
+        .map((element) => `COALESCE(${element}, 0)`)
+        .join(' + ')}) <= 100)`,
+      [],
+      'elemental_percent_check',
+    );
+    table.check(
+      `(COALESCE(${compounds.join(
+        ', ',
+      )}) IS NULL AND molecular_compounds_unit IS NULL) OR (COALESCE(${compounds.join(
+        ', ',
+      )}) IS NOT NULL AND molecular_compounds_unit IS NOT NULL)`,
+      [],
+      'molecular_compounds_unit_check',
+    );
+  });
+
+  // TODO: Migrate fertilizer table into here?
+  // Insert blank record into soil amendment product
+  await knex.raw(`
+    INSERT INTO soil_amendment_product (product_id)
+    SELECT product_id FROM product
+    WHERE product.type = 'soil_amendment_task'
+  `);
+
+  // Create amendment purpose table
+  await knex.schema.createTable('soil_amendment_purpose', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  // Add purpose types
+  for (const key of amendmentProductPurposeKeys) {
+    await knex('soil_amendment_purpose').insert({
+      key,
+    });
+  }
+
+  // Create amendment task method table
+  await knex.schema.createTable('soil_amendment_method', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  // Add method types
+  for (const key of amendmentMethodKeys) {
+    await knex('soil_amendment_method').insert({
+      key,
+    });
+  }
+
+  // Create new product table
+  await knex.schema.createTable('soil_amendment_task_products', (table) => {
+    table.increments('id').primary();
+    table.integer('task_id').references('task_id').inTable('task').notNullable();
+    // TODO: LF-4246 backfill data if nulls exist on prod and constrain this to notNullable()
+    table.integer('product_id').references('product_id').inTable('product');
+    table.decimal('weight', 36, 12);
+    table.enu('weight_unit', weightUnits);
+    table.decimal('volume', 36, 12);
+    table.enu('volume_unit', volumeUnits);
+    table.enu('application_rate_weight_unit', applicationRateWeightUnits);
+    table.enu('application_rate_volume_unit', applicationRateVolumeUnits);
+    // TODO: LF-4246 backfill data for percent_of_location_amended then make notNullable and defaulted to 100
+    table.decimal('percent_of_location_amended', 36, 12);
+    // TODO: LF-4246 backfill data for total_area_amended_in_m2 then make notNullable
+    table.decimal('total_area_amended_in_m2', 36, 12);
+    table.boolean('deleted').notNullable().defaultTo(false);
```

Comment:
Is `deleted` going to be `true` when a product is removed from the task?
If so, I don't think it's being taken care of...

Reply:
Yes that is correct. I think Joyce pointed out that the user story is for adding  a new soil amendment task. If this user story covers those cases too I should maybe add a ticket somewhere ya.

The edit cases I was trying to consider in the db architecture, but I have not thought about them in the controllers yet.

Reply:
I think the user story being only for add was in error -- if I remember correctly, I thought Loïc said he was going to update it to include read-only and edit. But he might have meant adding the screens in Figma when I hoped the ticket would be updated too 🤔 

So just to to make sure I'm understanding correctly, in either case this means everything related to PATCH is going to go on a new sub-task ticket, and not this one?

Reply:
Would that make sense to move this ticket through and keep the scope limited -ish. Haha I can take it on but just not to have blocker for other tickets to test out add function.

Reply:
I vote for another ticket, I'm full! ☺️

Reply:
https://lite-farm.atlassian.net/browse/LF-4260